### PR TITLE
Support customized PVC annotations from El Carro API

### PR DESCRIPTION
**Pull request by yian-zhou**
Some storage drivers like Trident use PVC annotations to provision PVs, e.g., read-only or R/W access permissions, etc. Those annotations vary for different usages. So having a way for customers to provide their own customized PVC annotations is needed for supporting such cases.

Change-Id: I802cb8501a2cc3b1cc046df6890729d36cda5a61

### CODE REVIEW

File: common/api/v1alpha1/disk.go [hunk 1]
```diff
@@ -37,6 +37,11 @@ type DiskSpec struct {
 	// for disk provisioning.
 	// +optional
 	StorageClass string `json:"storageClass,omitempty"`
+
+	// A map of string keys and values that can be used by external tooling to
+	// store and retrieve for the disk PVC.
```
Comment (ankel) [inline common/api/v1alpha1/disk.go:42]: nit: small reword on the doc text.

```suggestion
	// A map of string keys and values to be stored in the annotations of the PVC. These can be read and write by external tools through Kubernetes.
```

**Review by ankel — DISMISSED** [review]
Small reword on the doc text, otherwise LGTM.

File: common/api/v1alpha1/disk.go (hunk 1 above)
Comment (yian-zhou) [reply common/api/v1alpha1/disk.go]: Done. Thanks!

**Pull request merged**